python/cugraph/cugraph/community/egonet.py

```diff
@@ -95,6 +95,13 @@ def ego_graph(G, n, radius=1, center=True, undirected=None, distance=None):
 
     result_graph = type(G)(directed=G.is_directed())
 
+    if not G.edgelist.weights:
+        warning_msg = (
+            "'Ego_graph' requires the input graph to be weighted: Unweighted "
+            "graphs will not be supported in the next release."
+        )
+        warnings.warn(warning_msg, PendingDeprecationWarning)
```

Comment:
I think this should just be deprecated (not pending deprecated) if we know now that they won't be supported in the next release.  Plus there's [some debate](https://groups.google.com/g/dev-python/c/YFUIh_9KSYY) on the usefulness and the future of `PendingDeprecatedWarning` , but it seems like we want to go with `DeprecationWarning` anyway.

Reply:
I believe @jnke2016 removed this warning entirely.  The C API does support weighted graphs for this algorithm, but it will crash if you try to request weights from the result if the graph doesn't have weights.  We will add (in 23.02) a fix that returns a `NULL` pointer so that the python can check the result instead of having the call crash.